addons/web_editor/controllers/main.py

```diff
@@ -281,15 +286,51 @@ def add_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=Fa
             except ValueError as e:
                 return {'error': e.args[0]}
 
+        preprocessed_data.update({
+            'name': name,
+            'data': data,
+        })
+        return preprocessed_data
+
+    @http.route('/web_editor/media/add_data', type='json', auth='user', methods=['POST'], website=True)
+    def add_media_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=False, res_model='ir.ui.view', **kwargs):
+        """Creates a `web_editor.media` and its related `ir.attachment`."""
+        preprocessed_data = self._preprocess_data(name, data, is_image, quality, width, height)
+        if 'error' in preprocessed_data:
+            return preprocessed_data
+
+        self._clean_context()
+        media = self._media_create(
+            name=preprocessed_data['name'],
+            data=preprocessed_data['data'],
+            mimetype=preprocessed_data.get('mimetype'),
+            res_id=res_id,
+            res_model=res_model
+        )
+        return media._get_media_info()
+
+    @http.route('/web_editor/attachment/add_data', type='json', auth='user', methods=['POST'], website=True)
```

Comment:
Mmh, I did not expect we would have to keep this one for two use cases, maybe we should not:

1. Media dialog "Files" -> Don't you think this should use the new `web_editor.media` model too? Maybe with a field `type` to distinguish Images, Files and maybe later Videos ? When we'll implement the favorite and/or tags stuff, we'll probably want the same for files, no? (even if it is later move as a link dialog feature rather than a media dialog feature). 

2. Logo upload during website configurator flow, can't it be also a new `web_editor.media` record. Really, the logo should always be suggested in the media dialog whatever page or record you are.

What do you think?

Also cc @rdeodoo in case he does not agree :)

Reply:
:eyes: 
I've got to test things to answer here. I remember that things weren't working out of the box, but I don't remember the specifics and why I didn't implement it there.

Reply:
> Media dialog "Files" -> Don't you think this should use the new web_editor.media model too? Maybe with a field type to distinguish Images, Files and maybe later Videos ?

I'm not sure the `type` field is needed, we can always check the mimetype on the `attachment`.
But here, files through the media dialog _are_ saved media, that's not the problem.
That one route is in knowledge, with a use spanning between knowledge and modules with an html_field, and it triggers a `macro` which I can't wrap my head around:
1) the attachment / media exists in a knowledge article
2) by clicking on a button, it is created and linked to the other module's view
3) it is used in the chatter => that's the part that fails, because it's not looking for a media but for an attachment, with the `res_id` / `res_model` that are now stored on the media. Except I haven't found yet where that happens, and I'm not convinced it should be anything else than an attachment anyway.

> Logo upload during website configurator flow, can't it be also a new web_editor.media record. Really, the logo should always be suggested in the media dialog whatever page or record you are.

The `website.logo` is already a binary field itself. At first glance, it looks like I can modify the route, but the logo won't appear in the media dialog anyway (it's already not the case for now), because upon save, the attachment's `res_model` / `res_id` are overridden with `website.logo`. Which means that, by this point, there is a zombie `web_editor.media` not attached to any attachment anymore (I guess I could add some sort of autovacuum?).

Maybe there is also a test somewhere that doesn't like it, but I'm not sure as I can't find any by just checking the code. I guess that if I do it, I'll see when I push next time. ^^

Reply:
> But here, files through the media dialog are saved media, that's not the problem.
> That one route is in knowledge,

Oh, yes, I probably saw "File" and supposed it was the documents section of the media dialog without seeing I was in knowledge.

> with a use spanning between knowledge and modules with an html_field, and it triggers a macro which I can't wrap my head around

I let you investigate, if really you don't find, we'll talk about it IRL. In any case, this route should not exist anymore, at least not in web_editor.

> but the logo won't appear in the media dialog anyway (it's already not the case for now)

That's something that should change both for the sake of this task and from a functional point of view.
1. Maybe we should simply open the media dialog when asking the user for its logo. Debatable, probably useless from a functional point of view but who knows.
2. In any case, what should happen after the user choice:
  - A web_editor.media record is created for the logo. Its res_model+res_id+res_field can all be empty IMO. Such records with empty values for res_XXX should always be listed. Like all res_model=products + res_id=None ones should be listed on all products IMO. Like all res_model=None + res_field=something should be listed when editor a field named something IMO. Etc.
  - An ir.attachment should be created (automatically I guess?) to be linked to the website.logo field. Its original_id should be the web_editor.media.